guide/src/async-await.md

```diff
@@ -0,0 +1,242 @@
+# Async / Await
+
+Both Python and Rust have support for async functions, but bindings for these functions are not as 
+straightforward as they are for blocking functions. Both languages have a fairly distinct model for 
+async functions and Python's needs in particular can sometimes be restrictive.
+
+[pyo3-asyncio](https://github.com/awestlake87/pyo3-asyncio) was created to provide conversions 
+between async Python and async Rust as well as manage the nitty gritty details of Python's event 
+loop.
+
+## Awaiting an Async Python Function in Rust
+
+Let's take a look at a dead simple async Python function:
+
+```python
+async def py_sleep():
+    await asyncio.sleep(1)
+```
+
+This function simply sleeps for 1 second and returns. So what does this look like to PyO3? 
```

Comment:
> So what does this look like to PyO3? 

Isn't it better to first describe async function in Python, and then throw this question?

Reply:
My intention here was to somehow lead into the explanation of the async function, so maybe I should word it a bit differently.

The point that I was trying to make in this section is that even though async functions and normal functions are different, calling them via PyO3 is the same, you just have to treat the return value differently. I didn't want to dive too far into the specifics for awaitables and coroutines, just enough to explain the how/why for the `pyo3_asyncio::into_future` function.

Any tips on how I should reword this section? Should I just skip this line and go straight into the explanation in the next few paragraphs?

Reply:
> The point that I was trying to make in this section is that even though async functions and normal functions are different, calling them via PyO3 is the same, you just have to treat the return value differently. 

That makes sense, thanks.

> Should I just skip this line and go straight into the explanation in the next few paragraphs?

I don't think we should remove the whole paragraph, but packing the line into the next paragraph would make the document easier to follow.
E.g.,
```
This function simply sleeps for 1 second and returns a `coroutine` object. ...
```